sympy/series/sequences.py

```diff
@@ -0,0 +1,1123 @@
+    >>> s.formula
+    n**2
+
+    For value at a particular point
+
+    >>> s.coeff(3)
+    9
+
+    supports slicing
+
+    >>> s[:]
+    [0, 1, 4, 9, 16, 25]
+
+    iterable
+
+    >>> list(s)
+    [0, 1, 4, 9, 16, 25]
+
+    changing step size
+
+    >>> SeqFormula(n**2, (0, 5, 2))[:]
+    [0, 4, 16]
+
+    sequence starts from negative infinity
+
+    >>> SeqFormula(n**2, (-oo, 0))[0:6]
+    [0, 1, 4, 9, 16, 25]
+
+    See Also
+    ========
+
+    sympy.series.sequences.SeqPer
+    sympy.series.sequences.SeqFunc
+    """
+
+    @staticmethod
+    def _validate(formula):
+        """
+        Examples
+        ========
+
+        >>> from sympy.series.sequences import SeqFormula as seq
+        >>> from sympy.abc import n, m
+        >>> seq._validate((n**2, n))
+        (n**2, n)
+        >>> seq._validate(n**2)
+        (n**2, n)
+        >>> seq._validate((m*n**2, n))
+        (m*n**2, n)
+        >>> seq._validate(1)
+        (1, _k)
+        """
+        formula = sympify(formula)
+        if not is_sequence(formula, Tuple):
+            free = formula.free_symbols
+            if len(free) == 0:
+                k = Dummy('k')
+                formula = Tuple(formula, k)
+            elif len(free) == 1:
+                formula = Tuple(formula, free.pop())
+            else:
+                raise ValueError(
+                    " specify dummy variables for %s. If the formula contains"
+                    " more than one free symbol, a dummy variable should be"
+                    " supplied explicitly e.g., SeqFormula((m*n**2, n), (0, 5))"
+                    % formula)
+        return formula
+
+    @property
+    def formula(self):
+        return self.gen[0]
+
+    @property
+    def variables(self):
+        """Bounded variables
+        In the case of SeqFormula
+        it will be a tuple with only one symbol
+        """
+        return (self.gen[1],)
+
+    def _eval_coeff(self, pt):
+        d = self.variables[0]
+        return self.formula.subs(d, pt)
+
+    def _add(self, other):
+        """See docstring of SeqBase._add"""
+        if isinstance(other, SeqFormula):
+            form1, v1 = self.formula, self.variables[0]
+            form2, v2 = other.formula, other.variables[0]
+            formula = (form1 + form2.subs(v2, v1), v1)
+            return SeqFormula(formula, self._intersect_interval(other))
+
+
+    def _mul(self, other):
+        """See docstring of SeqBase._mul"""
+        if isinstance(other, SeqFormula):
+            form1, v1 = self.formula, self.variables[0]
+            form2, v2 = other.formula, other.variables[0]
+            formula = (form1 * form2.subs(v2, v1), v1)
+            return SeqFormula(formula, self._intersect_interval(other))
+
+    def coeff_mul(self, coeff):
+        """See docstring of SeqBase.coeff_mul"""
+        coeff = sympify(coeff)
+        formula = self.formula * coeff
+        return SeqFormula((formula, self.variables[0]), \
```

Comment:
As a note @leosartaj, you can check for stuff like this with a PEP 8 checker like flake8. Don't feel obligated to change everything it flags (especially long lines, error 501), but it will catch things like this, `is None`, etc.